### PR TITLE
Compat macro for taking strings or paths

### PR DESCRIPTION
**Pull request by rofinn**

### CODE REVIEW

File: src/compat.jl [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+    push!(body, :(return result))
+
+    fdef[:body].args = body
+
+    return MacroTools.combinedef(fdef)
+end
+
+function _ispath(t::Symbol)
+    T = eval(t)
```
Comment (oxinabox) [inline src/compat.jl]: Safer: `T=getfield(mod, t)`
Where `mod` is passed in to the function from the macro and is the calling module `__module__`


This A) avoids calling `eval` which could do anything
B) runs with the names that are in scope at the calling location.
Which could include 3rd party `AbstractPath` types

Comment (rofinn) [reply src/compat.jl]: Good call 👍

**Comment by rofinn** [comment]
Hmmm, coverage seems to be a false alarm because the lines not hit would have caused the tests to fail.

**Comment by rofinn** [comment]
Refactoring seems to have fixed this.

**Pull request merged**